packages/check-time/extra/check-time.c

```diff
@@ -0,0 +1,41 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/timex.h>
+#include <string.h>
+#include <errno.h>
+
+#define USEC_PER_MSEC 1000L
+#define MAX_EST_ERROR_US 100L * USEC_PER_MSEC /* 100 millisecond */
+
+int main() {
+	struct timex tx;
+	int rc;
+	long long int error;
+
+	if (getenv("IGNORE_CHECK_TIME")) {
+		fprintf(stderr, "check-time ignoring time, passing no matter what\n");
+		return 0;
+	}
+
+	memset(&tx, 0, sizeof(tx));
+	rc = adjtimex(&tx);
+	error = tx.esterror - MAX_EST_ERROR_US;
+
+	if (rc == TIME_BAD)
+		fprintf(stderr, "Time is marked as bad\n");
+	else if (rc == -1)
+		perror("adjtimex");
```

Comment:
referring to @spahl. From my perspective, next person who touches this will probably end up breaking it and not meaning to. This also has the dangling else parse problem / the else statements could re-associate in fun ways you wouldn't expect with minor changes to the code.